File: scripts/send-test-progress.js
#!/usr/bin/env node

const WEBHOOK_URL = process.env.WEBHOOK_URL || "http://localhost:3000/api/tasks/update";
const EXTERNAL_TASK_ID = process.env.EXTERNAL_TASK_ID || "claude_task_001";

const basePayload = {
  externalTaskId: EXTERNAL_TASK_ID,
  aiToolName: "Claude",
  title: "Landing page copy generation",
  description: "Generate first draft for product landing page",
  status: "in_progress",
  progress: 65,
  currentStage: "Writing first draft",
  log: "Claude is generating section headlines",
};

const updates = [
  {
    status: "not_started",
    progress: 0,
    currentStage: "Queued",
    log: "Task created and waiting to start",
  },
  {
    status: "in_progress",
    progress: 35,
    currentStage: "Collecting references",
    log: "Claude is gathering source material",
  },
  {
    status: "in_progress",
    progress: 65,
    currentStage: "Writing first draft",
    log: "Claude is generating section headlines",
  },
  {
    status: "completed",
    progress: 100,
    currentStage: "Draft completed",
    log: "First draft delivered successfully",
  },
];

const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

async function sendUpdate(stepUpdate, stepIndex) {
  const payload = { ...basePayload, ...stepUpdate, externalTaskId: EXTERNAL_TASK_ID };
  console.log(`[${stepIndex}] Sending update: status=${payload.status} progress=${payload.progress}%`);

  const res = await fetch(WEBHOOK_URL, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });

  if (!res.ok) {
    const text = await res.text();
    throw new Error(`HTTP ${res.status}: ${text}`);
  }

  const body = await res.json();
  console.log(`    -> OK (${res.status}) taskId=${body?.task?.id ?? "n/a"}`);
}

async function main() {
  console.log(`Webhook URL: ${WEBHOOK_URL}`);
  console.log(`External Task ID: ${EXTERNAL_TASK_ID}`);

  for (let i = 0; i < updates.length; i += 1) {
    await sendUpdate(updates[i], i + 1);
    if (i < updates.length - 1) {
      await sleep(1200);
    }
  }

  console.log("Done. Open the dashboard and refresh/auto-sync to see updates.");
}

main().catch((err) => {
  console.error("Failed to send updates:", err.message);
  process.exit(1);
});

File: scripts/send-test-progress.py
#!/usr/bin/env python3
"""
Send demo task progress updates to the local webhook endpoint.

Requires:
  pip install requests
"""

from __future__ import annotations

import os
import time
from copy import deepcopy

import requests

WEBHOOK_URL = os.getenv("WEBHOOK_URL", "http://localhost:3000/api/tasks/update")
EXTERNAL_TASK_ID = os.getenv("EXTERNAL_TASK_ID", "claude_task_001")

BASE_PAYLOAD = {
    "externalTaskId": EXTERNAL_TASK_ID,
    "aiToolName": "Claude",
    "title": "Landing page copy generation",
    "description": "Generate first draft for product landing page",
    "status": "in_progress",
    "progress": 65,
    "currentStage": "Writing first draft",
    "log": "Claude is generating section headlines",
}

UPDATES = [
    {
        "status": "not_started",
        "progress": 0,
        "currentStage": "Queued",
        "log": "Task created and waiting to start",
    },
    {
        "status": "in_progress",
        "progress": 35,
        "currentStage": "Collecting references",
        "log": "Claude is gathering source material",
    },
    {
        "status": "in_progress",
        "progress": 65,
        "currentStage": "Writing first draft",
        "log": "Claude is generating section headlines",
    },
    {
        "status": "completed",
        "progress": 100,
        "currentStage": "Draft completed",
        "log": "First draft delivered successfully",
    },
]


def send_update(step_payload: dict, step_idx: int) -> None:
    payload = deepcopy(BASE_PAYLOAD)
    payload.update(step_payload)
    payload["externalTaskId"] = EXTERNAL_TASK_ID

    print(f"[{step_idx}] Sending update: status={payload['status']} progress={payload['progress']}%")
    response = requests.post(WEBHOOK_URL, json=payload, timeout=10)
    response.raise_for_status()
    body = response.json()
    print(f"    -> OK ({response.status_code}) taskId={body.get('task', {}).get('id', 'n/a')}")


def main() -> None:
    print(f"Webhook URL: {WEBHOOK_URL}")
    print(f"External Task ID: {EXTERNAL_TASK_ID}")

    for idx, update in enumerate(UPDATES, start=1):
        send_update(update, idx)
        if idx < len(UPDATES):
            time.sleep(1.2)

    print("Done. Open the dashboard and refresh/auto-sync to see updates.")


if __name__ == "__main__":
    main()

